recipes/vulkan-validationlayers/all/conanfile.py

```diff
@@ -57,14 +61,18 @@ def _get_compatible_spirv_tools_version(self):
         }.get(str(self.version), False)
 
     def requirements(self):
-        self.requires("spirv-tools/{}".format(self._get_compatible_spirv_tools_version), private=True)
+        #TODO set private=False, once the issue is resolved https://github.com/conan-io/conan/issues/9390
+        set_private = not self._is_cross_building()
+        if tools.is_apple_os(self.settings.os) and self.settings.arch in ["armv8", "armv8_32", "armv8.3"]:
+            set_private = False
+        self.requires("spirv-tools/{}".format(self._get_compatible_spirv_tools_version), private=set_private)
```

Comment:
The issue is not about cross-build but 2 profiles, so I guess your detection of 2 profiles is wrong.

Reply:
So, how to detect two profiles?

Reply:
```suggestion
        # TODO: set private=True, once the issue is resolved https://github.com/conan-io/conan/issues/9390
        self.requires("spirv-tools/{}".format(self._get_compatible_spirv_tools_version), private=not hasattr(self, "settings_build"))
```